src/rust/engine/options/src/arg_splitter.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+
```

Comment:
The logic in this file is a fairly straight port of the Python logic, if that is helpful in reviewing. More importantly, the Python tests pass (in a branch) with this wired in...

Reply:
The Python equivalent seems to also be managing flags (`scope_to_flags`), but that's not here. How does that fit together?

Reply:
The Python equivalent no longer manages flags, see https://github.com/pantsbuild/pants/pull/21811